### PR TITLE
Remove the reserved macro names

### PR DESCRIPTION
**Pull request by stof**
Replaces #1765

### CODE REVIEW

File: lib/Twig/Parser.php [hunk 1]
```diff
@@ -254,28 +253,17 @@ public function hasMacro($name)
...
     {
-        if ($this->isReservedMacroName($name)) {
-            throw new Twig_Error_Syntax(sprintf('"%s" cannot be used as a macro name as it is a reserved keyword', $name), $node->getLine(), $this->getFilename());
-        }
-
         $this->macros[$name] = $node;
     }
 
+    /**
+     * @deprecated since 1.21. Will be removed in 2.0. There is no reserved macro names anymore
```
Comment (fabpot) [inline lib/Twig/Parser.php]: I would not even bother with that and remove the method altogether.

**Comment by hason** [comment]
Great, this change I suggested several times #1310 ([commit](https://github.com/hason/Twig/commit/33aa7de9b3bd3cdfa5a0c8892500e82b26d055a2)). This is BC break (see #1217) and I vote :+1: for Twig 2.0 or Twig 3.0.

We really should resolve macros, especially #1212 and #1217.

**Comment by stof** [comment]
arf, I hate the fact that we modify the context when importing macros, as the context propagates to other templates.

**Comment by hason** [comment]
@fabpot What do you think? I think we should explicitly import all macros that are also imported in parent templates. Will this change in version 2 or 3?

**Comment by fabpot** [comment]
We cannot really have this one in 1.x because of the BC break mentioned by @hason, even if the behavior in #1217 should not be relied on. So, I propose to merge it in master (sorry @stof, you will have to re-open yet a new one) and document the BC break in case some people relied on the current behavior.

**Comment by fabpot** [comment]
By the way, using `__macro__` is probably "safer".

**Comment by stof** [comment]
@fabpot why safer ? We are already using ``block_`` as prefix for blocks.

**Comment by stof** [comment]
OK, closing and reopening it on master

**Pull request closed without merging**